integration-tests/src/tests/client/process_blocks.rs

```diff
@@ -4511,3 +4511,162 @@ mod contract_precompilation_tests {
+    fn compare_node_touches() {
+        let mut genesis =
+            Genesis::test(vec!["test0".parse().unwrap(), "test1".parse().unwrap()], 1);
+        let epoch_length = 10;
+        let num_blocks = 5;
+
+        #[cfg(feature = "protocol_feature_chunk_nodes_cache")]
+        let old_protocol_version = ProtocolFeature::ChunkNodesCache.protocol_version() - 1;
+        #[cfg(not(feature = "protocol_feature_chunk_nodes_cache"))]
+        let old_protocol_version = PROTOCOL_VERSION - 1;
+
+        genesis.config.epoch_length = epoch_length;
+        genesis.config.protocol_version = old_protocol_version;
+        let chain_genesis = ChainGenesis::from(&genesis);
+        let mut env = TestEnv::builder(chain_genesis)
+            .runtime_adapters(create_nightshade_runtimes(&genesis, 1))
+            .build();
+
+        deploy_test_contract(
+            &mut env,
+            "test0".parse().unwrap(),
+            near_test_contracts::rs_contract(),
+            num_blocks,
+            1,
+        );
+
+        let signer = InMemorySigner::from_seed("test0".parse().unwrap(), KeyType::ED25519, "test0");
+        let tx_node_touches: Vec<u64> = (0..4)
+            .map(|i| {
+                let base_cost = RuntimeConfig::test().wasm_config.ext_costs.touching_trie_node;
+
+                let tx_hash = if i < 1 {
+                    process_transaction(&mut env, &signer, num_blocks, old_protocol_version)
+                } else {
+                    process_transaction(
+                        &mut env,
+                        &signer,
+                        2 * epoch_length,
+                        old_protocol_version + 1,
+                    )
+                };
+
+                let final_result =
+                    env.clients[0].chain.get_final_transaction_result(&tx_hash).unwrap();
+                assert!(matches!(final_result.status, FinalExecutionStatus::SuccessValue(_)));
+                let transaction_outcome =
+                    env.clients[0].chain.get_execution_outcome(&tx_hash).unwrap();
+                let receipt_ids = transaction_outcome.outcome_with_id.outcome.receipt_ids;
+                assert_eq!(receipt_ids.len(), 1);
+                let receipt_execution_outcome =
+                    env.clients[0].chain.get_execution_outcome(&receipt_ids[0]).unwrap();
+                let metadata = receipt_execution_outcome.outcome_with_id.outcome.metadata.clone();
+                let touching_trie_node_cost = match metadata {
+                    ExecutionMetadata::V1 => panic!("ExecutionMetadata cannot be empty"),
+                    ExecutionMetadata::V2(profile_data) => {
+                        profile_data.get_ext_cost(ExtCosts::touching_trie_node)
+                    }
+                };
+
+                touching_trie_node_cost / base_cost
+            })
+            .collect();
+
+        assert_eq!(tx_node_touches[1], 12);
+        #[cfg(feature = "protocol_feature_chunk_nodes_cache")]
+        {
```

Comment:
The third assert does it. If that was the case, the number of touches there would be 4 instead of 8, because we would not count first 4 nodes **both times**.
I will clarify this in comments as well.